cmd/satellite/main.go

```diff
@@ -178,11 +180,19 @@ func processStats(stats [][]uplink.GPUStatSample) []uplink.GPUStatSample {
 	return stats[len(stats)-1]
 }
 
-func setGPUHandler(isFakeGPUs bool) gpustats.GPUDataSource {
+func setGPUHandler(log *slog.Logger, isFakeGPUs bool) gpustats.GPUDataSource {
 	if isFakeGPUs {
 		return gpustats.FakeGPU{}
 	} else {
-		return gpustats.NvidiaGPUHandler{}
+		passwdfile, err := os.Open("/etc/passwd")
+		passwd, err1 := passwd.Parse(passwdfile)
```

Comment:
If this open fails, we should bail early, as `passwdfile` may be a `nil` `io.Reader`, and we'll die